quic/s2n-quic/src/connection/handle.rs

```diff
@@ -378,13 +378,13 @@ macro_rules! impl_handle_api {
             query.into()
         }
 
-        pub fn datagram_mut<Query, SenderType, Outcome>(
+        pub fn datagram_mut<Query, ProviderType, Outcome>(
```

Comment:
can you add a comment on what this function expects: `Receiver` and `Sender`

you can take a look at query function above for some example